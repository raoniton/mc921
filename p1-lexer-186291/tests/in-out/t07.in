class Main {
    public static void main(String[] args) {
        int n, reverse = 0, rem;
        n = 1000;
        while (n != 0) {
            rem = n % 10;
            reverse = reverse * 10 + rem;
            n = n / 10;
        }
        print("Reversed Number: ", reverse);
    }
}