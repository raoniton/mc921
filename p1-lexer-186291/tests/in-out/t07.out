LexToken(CLASS,'class',1,0)
LexToken(ID,'Main',1,6)
LexToken(LBRACE,'{',1,11)
LexToken(PUBLIC,'public',2,17)
LexToken(STATIC,'static',2,24)
LexToken(VOID,'void',2,31)
LexToken(MAIN,'main',2,36)
LexToken(LPAREN,'(',2,40)
LexToken(STRING,'String',2,41)
LexToken(LBRACKET,'[',2,47)
LexToken(RBRACKET,']',2,48)
LexToken(ID,'args',2,50)
LexToken(RPAREN,')',2,54)
LexToken(LBRACE,'{',2,56)
LexToken(INT,'int',3,66)
LexToken(ID,'n',3,70)
LexToken(COMMA,',',3,71)
LexToken(ID,'reverse',3,73)
LexToken(ASSIGN,'=',3,81)
LexToken(INT_LITERAL,'0',3,83)
LexToken(COMMA,',',3,84)
LexToken(ID,'rem',3,86)
LexToken(SEMI,';',3,89)
LexToken(ID,'n',4,99)
LexToken(ASSIGN,'=',4,101)
LexToken(INT_LITERAL,'1000',4,103)
LexToken(SEMI,';',4,107)
LexToken(WHILE,'while',5,117)
LexToken(LPAREN,'(',5,123)
LexToken(ID,'n',5,124)
LexToken(NE,'!=',5,126)
LexToken(INT_LITERAL,'0',5,129)
LexToken(RPAREN,')',5,130)
LexToken(LBRACE,'{',5,132)
LexToken(ID,'rem',6,146)
LexToken(ASSIGN,'=',6,150)
LexToken(ID,'n',6,152)
LexToken(MOD,'%',6,154)
LexToken(INT_LITERAL,'10',6,156)
LexToken(SEMI,';',6,158)
LexToken(ID,'reverse',7,172)
LexToken(ASSIGN,'=',7,180)
LexToken(ID,'reverse',7,182)
LexToken(TIMES,'*',7,190)
LexToken(INT_LITERAL,'10',7,192)
LexToken(PLUS,'+',7,195)
LexToken(ID,'rem',7,197)
LexToken(SEMI,';',7,200)
LexToken(ID,'n',8,214)
LexToken(ASSIGN,'=',8,216)
LexToken(ID,'n',8,218)
LexToken(DIVIDE,'/',8,220)
LexToken(INT_LITERAL,'10',8,222)
LexToken(SEMI,';',8,224)
LexToken(RBRACE,'}',9,234)
LexToken(PRINT,'print',10,244)
LexToken(LPAREN,'(',10,249)
LexToken(STRING_LITERAL,'"Reversed Number: "',10,250)
LexToken(COMMA,',',10,269)
LexToken(ID,'reverse',10,271)
LexToken(RPAREN,')',10,278)
LexToken(SEMI,';',10,279)
LexToken(RBRACE,'}',11,285)
LexToken(RBRACE,'}',12,287)
